src/plan/immix/mutator.rs

```diff
@@ -15,16 +15,7 @@ use crate::{
 };
 use enum_map::EnumMap;
 
-pub fn immix_mutator_prepare<VM: VMBinding>(mutator: &mut Mutator<VM>, _tls: VMWorkerThread) {
-    let immix_allocator = unsafe {
-        mutator
-            .allocators
-            .get_allocator_mut(mutator.config.allocator_mapping[AllocationSemantics::Default])
-    }
-    .downcast_mut::<ImmixAllocator<VM>>()
-    .unwrap();
-    immix_allocator.reset();
-}
+pub fn immix_mutator_prepare<VM: VMBinding>(_mutator: &mut Mutator<VM>, _tls: VMWorkerThread) {}
```

Comment:
Is it possible to not call this function, instead of making it a no-op?

Reply:
This function is called in the `PrepareMutator` work packet.  `PrepareMutator` simply calls `mutator.prepare` which calls `(*self.config.prepare_func)(self, tls)`.  We can avoid calling `prepare_func` by changing `prepare_func` to an `Option<...>` and test if it is `Some(...)` before calling.  Alternatively, we can assign a dedicated no-op function to `prepare_func` and test if it points to the no-op function before calling.  But I think both methods are too complicated.  Because `MutatorConfig::prepare_func` is only called once per GC, it is not worth eliminating this function call.

Reply:
> Because MutatorConfig::prepare_func is only called once per GC, it is not worth eliminating this function call.

This is what I was worrying about. Some Google internal workloads can have over 1000 mutator threads...

Reply:
> > Because MutatorConfig::prepare_func is only called once per GC, it is not worth eliminating this function call.
> 
> This is what I was worrying about. Some Google internal workloads can have over 1000 mutator threads...

Did you find `Release` and `Prepare` takes up a considerable proportion of the total time spent in GC in those cases?

Reply:
> This is what I was worrying about. Some Google internal workloads can have over 1000 mutator threads...

Honestly they should use OpenJDK 21 and Virtual Threads.  But if they insist on using native threads, we may try to eliminate the need to create the `PrepareMutator` work packet.  We can add a per-plan constant, or a field in `PlanConstraint`, like `need_prepare_mutator: bool`.  If it is false, we won't create the `PrepareMutator` work packet in the first place.

Reply:
> > This is what I was worrying about. Some Google internal workloads can have over 1000 mutator threads...
> 
> Honestly they should use OpenJDK 21 and Virtual Threads. But if they insist on using native threads, we may try to eliminate the need to create the `PrepareMutator` work packet. We can add a per-plan constant, or a field in `PlanConstraint`, like `need_prepare_mutator: bool`. If it is false, we won't create the `PrepareMutator` work packet in the first place.

Having a constant in `PlanConstraint` and avoiding the work packet creation sounds like a good idea.